CONTRIBUTING.md

```diff
@@ -74,8 +70,58 @@ BREAKING CHANGE: numpy 1.0.2 no longer supported
 Refs: #123
 ```
 
-Code
-------
+### Breaking changes and deprecation
+
+Since we are still pre-1.0, [SemVer] states that any release may contain breaking
+changes. However, breaking changes should not be made without warning.
+
+**Any breaking changes must have a deprecation period of at least one minor release**
+before the breaking change is made. If the change is one that may require significant
+changes to client code, such as removing a function or class entirely, the deprecation
+period must instead be at least **two minor releases**.
+
+Ensure that during the deprecation period, the old behaviour still works, but raises a
+`DeprecationWarning` with an appropriate message. If at all possible, ensure that there
+is straightforward signposting for how users should change their code to use
+non-deprecated parts of the codebase instead.
+
+As an example, this is what the deprecation period for renaming `my_old_function` to
+`my_new_function` would look like.
+
+```python
+# v0.1.0:
+def my_old_function(x: int) -> int:
+    return x + x + x + x
+
+# v0.2.0:
+def my_new_function(x: int) -> int:
+    return x*4
+
+@deprecated("Renamed to my_new_function; will be removed in v0.3.0")
+def my_old_function(x: int) -> int:
+    return my_new_function(x)
+
+# v0.3.0:
+def my_new_function(x: int) -> int:
+    return x*4
+```
+
+Examples of breaking changes include, but are not limited to:
```

Comment:
I think these examples are bloating the file. I'd delete the rest of this section.